pkg/cmd/cli/cmd/newapp.go

```diff
@@ -243,16 +243,26 @@ func (o *NewAppOptions) Run() error {
 		return err
 	}
 
-	// if the user has set the "app" label explicitly on their objects in the template,
-	// we should not return a failure when we can't set it ourselves.
-	ignoreLabelFailure := false
-	if len(config.Labels) == 0 && len(result.Name) > 0 {
-		config.Labels = map[string]string{"app": result.Name}
```

Comment:
@jwforres so the existing behavior was to only add the app label if the user didn't specify any labels to add (config.Labels).